File: AtlasWeb/Core/Modules/Tibbi_Surec/Manipulasyon_Modulu/AppointmentFormManipulationViewModel.cs
//$6FEF3835
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class ManipulationServiceController
    {
        partial void PreScript_AppointmentFormManipulation(AppointmentFormManipulationViewModel viewModel, Manipulation manipulation, TTObjectContext objectContext)
        {
            viewModel.AppointmentDescription = BaseAction.GetFullAppointmentDescription(manipulation);
        }

        partial void PostScript_AppointmentFormManipulation(AppointmentFormManipulationViewModel viewModel, Manipulation manipulation, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            manipulation.CompleteMyNewAppoinments();
        }
    }
}

namespace Core.Models
{
    public partial class AppointmentFormManipulationViewModel
    {
        public string AppointmentDescription;
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Manipulasyon_Modulu/AppointmentFormManipulationViewModel.ts
//$6120141A
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { Manipulation } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';
export class AppointmentFormManipulationViewModel extends BaseViewModel {
    @Type(() => Manipulation)
    public _Manipulation: Manipulation = new Manipulation();

    public AppointmentDescription: string;
}
